test/unit/org/apache/cassandra/cql3/statements/TransactionStatementTest.java

```diff
@@ -483,6 +500,30 @@ public void shouldRejectInClauseInLet()
                 .hasMessageContaining(String.format(NO_PARTITION_IN_CLAUSE_WITH_LIMIT, "SELECT", "at"));
     }
 
+    @Test
+    public void shouldRejectInClauseInLetWithBind()
+    {
+        Select select = Select.builder()
+                              .table(tbl(1))
+                              .in("k", 0, 1)
+                              .limit(1)
+                              .build();
+
+        TransactionStatement stmt = (TransactionStatement) prepare(Txn.wrap(select).toCQL());
+        QueryState state = QueryState.forInternalCalls();
+        Dispatcher.RequestTime now = Dispatcher.RequestTime.forImmediateExecution();
+        Assertions.assertThatThrownBy(() -> stmt.execute(state, QueryOptions.forInternalCalls(Arrays.asList(select.bindsEncoded())), now)).isInstanceOf(InvalidRequestException.class)
+                  .hasMessageContaining(String.format(NO_PARTITION_IN_CLAUSE_WITH_LIMIT, "SELECT", "at"));
+        Txn txn = Txn.builder()
+                   .addLet("a", select)
+                   .addReturn(select)
```

Comment:
nit: Would it make sense to return something like `SELECT a.whatever` here instead of a standard SELECT? I guess that would ensure we're hitting the error because of the LET specifically (as there is a reference return)?

Reply:
switched to `a.v`